ocs_ci/ocs/ui/page_objects/buckets_tab.py

```diff
@@ -1,18 +1,146 @@
+            + ".txt"
+        )
+        filepath = os.path.join(folder_path, filename)
+
+        content = "".join(
+            random.choices(string.ascii_letters + string.digits + " \n", k=100)
+        )
+
+        with open(filepath, "w") as f:
+            f.write(content)
+
+        return folder_path
+
+    def create_bucket_ui(self, method):
+
+        self.do_click(self.bucket_tab["create_bucket_button"])
+        if method == "obc":
+            self.create_bucket_via_obc()
+        elif method == "s3":
+            self.create_bucket_via_s3()
+        else:
+            raise ValueError(f"Invalid method: {method}")
+
+    def create_bucket_via_obc(self):
+
+        self.do_click(self.bucket_tab["create_bucket_button_obc"])
+        self.do_click(self.bucket_tab["storage_class_dropdown"])
+        self.do_click(self.bucket_tab["storage_class_noobaa_option"])
+        self.do_click(self.bucket_tab["submit_button_obc"])
+
+    def create_bucket_via_s3(self):
+        name_generator = f"test-bucket-s3-{uuid.uuid4()}"
+
+        self.do_click(self.bucket_tab["create_bucket_button_s3"])
+        self.do_send_keys(self.bucket_tab["bucket_name_input"], name_generator)
+        self.do_click(self.bucket_tab["submit_button_obc"])
+
+    def create_folder_in_bucket(self, bucket_name=None, folder_name=None):
+        # Note That object must be uploaded to the folder before navigating out of the bucket,
+        # else the folder will be vanished
+
+        if bucket_name:
+            self.do_click(f"//tr//a[contains(text(), '{bucket_name}')]", By.XPATH)
+        else:
+            self.do_click(self.bucket_tab["first_bucket"])
+
+        if not folder_name:
+            folder_name = f"test-folder-{uuid.uuid4()}"
+
+        self.do_click(self.bucket_tab["create_folder_button"])
+        self.do_send_keys(self.bucket_tab["folder_name_input"], folder_name)
+        self.do_click(self.bucket_tab["submit_button_folder"])
+
+        folder_path = self.generate_folder_with_file()
+
+        time.sleep(2)  # Wait for folder navigation
+
+        logger.info("=== DEBUG: STARTING FILE UPLOAD ===")
+
+        try:
+            # Find the hidden file input
+            file_input = self.driver.find_element(
+                By.XPATH, "//input[@type='file'][@webkitdirectory]"
+            )
+            logger.info("Found directory input")
+
+            # Make the input visible and remove directory requirement
+            self.driver.execute_script(
+                """
+                arguments[0].removeAttribute('webkitdirectory');
+                arguments[0].removeAttribute('directory');
+                arguments[0].style.display = 'block';
+                arguments[0].style.visibility = 'visible';
+                arguments[0].style.height = '1px';
+                arguments[0].style.width = '1px';
+                arguments[0].style.opacity = '1';
+            """,
+                file_input,
+            )
```

Comment:
we don't use js scripts. It is not checking UX, because all the web elements may be hidden or overlayed but still be available for script.
remove `webkitdirectory` changes the behavior of the product, s we check not management-console, but modified management-console. 
arguments[0].removeAttribute('directory') same as above; this way we make input element accept files, not directories. 
Question: is something in UI hidden or not interactable via locators that forces us to use this script?

Reply:
> Question: is something in UI hidden or not interactable via locators that forces us to use this script?

As far as i could find, yes